src/app/choose-template/main-server-action.ts

```diff
@@ -0,0 +1,92 @@
+'use server';
+import { z } from 'zod';
+import { FormState, FormId } from '../../utils/types';
+import { zodValidationServerAction } from '../../utils/zod-validation-server-action';
+import { useRouter } from 'next/navigation';
+
+const serverActions: Partial<
+  Record<FormId, (formState: FormState, formData: FormData) => FormState>
+> = {
+  'choose-template': (formState: FormState, formData: FormData) =>
+    zodValidationServerAction(
+      formState,
+      formData,
+      z.object({
+        page: z.enum(
+          [
+            'create-sms-template',
+            'create-email-template',
+            'create-nhs-app-template',
+            'create-letter-template',
+          ],
+          { message: 'Select a template type' }
+        ),
+      })
+    ),
+  'create-nhs-app-template-back': (formState: FormState, formData: FormData) =>
+    zodValidationServerAction(
+      formState,
+      formData,
+      z.object({
+        nhsAppTemplateName: z.string({ message: 'Internal server error' }),
+        nhsAppTemplateMessage: z.string({ message: 'Internal server error' }),
+      }),
+      'choose-template'
+    ),
+  'create-nhs-app-template': (formState: FormState, formData: FormData) =>
+    zodValidationServerAction(
+      formState,
+      formData,
+      z.object({
+        nhsAppTemplateName: z
+          .string()
+          .min(1, { message: 'Enter a template name' }),
+        nhsAppTemplateMessage: z
+          .string()
+          .min(1, { message: 'Enter a template message' }),
```

Comment:
This should also have a max set to 5000. We can't rely on the client-side  input limiting the value to 5000. You can bypass the limit on the client by manually editing the HTML.

Reply:
updated